frontend/src/metabase/components/FieldValuesWidget/FieldValuesWidget.tsx

```diff
@@ -370,6 +372,30 @@ export function FieldValuesWidgetInner({
     search.current(value);
   };
 
+  const fieldValues = useMemo(() => {
```

Comment:
Might be worth turning this into a object/map for value -> label mapping to avoid N2 cycles